### ISSUE TITLE
vpc generation=2 multiple ibm_is_subnet in vpc fail

### ISSUE THREAD

**Issue by powellquiring** [IBM-Cloud/terraform-provider-ibm#768]
I've tried a similar construct using the cli and it worked ok

```
variable region {default = "us-south"}
variable ibmcloud_api_key {}

variable basename { default = "tfbug" }
variable zone {default = "us-south-1"}
provider "ibm" {
  region           = "${var.region}"
  ibmcloud_api_key = "${var.ibmcloud_api_key}"
  generation       = "2"
}

resource "ibm_is_vpc" "vpc" {
  name           = "tfbug"
}
resource "ibm_is_subnet" "backend" {
  name                     = "${var.basename}-backend-subnet"
  vpc                      = "${ibm_is_vpc.vpc.id}"
  zone                     = "${var.zone}"
#  public_gateway           = "${join("", ibm_is_public_gateway.backend.*.id)}"
  total_ipv4_address_count = 256
}
resource "ibm_is_subnet" "frontend" {
  name                     = "${var.basename}-front-subnet"
  vpc                      = "${ibm_is_vpc.vpc.id}"
  zone                     = "${var.zone}"
  total_ipv4_address_count = 256
}
resource "ibm_is_subnet" "bastion" {
  name                     = "${var.basename}-bastion-subnet"
  vpc                      = "${ibm_is_vpc.vpc.id}"
  zone                     = "${var.zone}"
  total_ipv4_address_count = 256
}

Terraform Version
~/tmp/tfsubnettest $ tf version
Terraform v0.11.14
+ provider.ibm v0.17.5
```
apply
```
~/tmp/tfsubnettest $ tfa
ibm_is_vpc.vpc: Creating...
  classic_access:          "" => "false"
  default_network_acl:     "" => "<computed>"
  default_security_group:  "" => "<computed>"
  is_default:              "" => "false"
  name:                    "" => "tfbug"
  resource_controller_url: "" => "<computed>"
  resource_crn:            "" => "<computed>"
  resource_group:          "" => "<computed>"
  resource_group_name:     "" => "<computed>"
  resource_name:           "" => "<computed>"
  resource_status:         "" => "<computed>"
  status:                  "" => "<computed>"
ibm_is_vpc.vpc: Still creating... (10s elapsed)
ibm_is_vpc.vpc: Creation complete after 14s (ID: r006-ed0e6556-7aa8-4a56-b247-094f09d1b2c4)
ibm_is_subnet.frontend: Creating...
  available_ipv4_address_count: "" => "<computed>"
  ip_version:                   "" => "ipv4"
  ipv4_cidr_block:              "" => "<computed>"
  ipv6_cidr_block:              "" => "<computed>"
  name:                         "" => "tfbug-front-subnet"
  network_acl:                  "" => "<computed>"
  resource_controller_url:      "" => "<computed>"
  resource_crn:                 "" => "<computed>"
  resource_group_name:          "" => "<computed>"
  resource_name:                "" => "<computed>"
  resource_status:              "" => "<computed>"
  status:                       "" => "<computed>"
  total_ipv4_address_count:     "" => "256"
  vpc:                          "" => "r006-ed0e6556-7aa8-4a56-b247-094f09d1b2c4"
  zone:                         "" => "us-south-1"
ibm_is_subnet.backend: Creating...
  available_ipv4_address_count: "" => "<computed>"
  ip_version:                   "" => "ipv4"
  ipv4_cidr_block:              "" => "<computed>"
  ipv6_cidr_block:              "" => "<computed>"
  name:                         "" => "tfbug-backend-subnet"
  network_acl:                  "" => "<computed>"
  resource_controller_url:      "" => "<computed>"
  resource_crn:                 "" => "<computed>"
  resource_group_name:          "" => "<computed>"
  resource_name:                "" => "<computed>"
  resource_status:              "" => "<computed>"
  status:                       "" => "<computed>"
  total_ipv4_address_count:     "" => "256"
  vpc:                          "" => "r006-ed0e6556-7aa8-4a56-b247-094f09d1b2c4"
  zone:                         "" => "us-south-1"
ibm_is_subnet.bastion: Creating...
  available_ipv4_address_count: "" => "<computed>"
  ip_version:                   "" => "ipv4"
  ipv4_cidr_block:              "" => "<computed>"
  ipv6_cidr_block:              "" => "<computed>"
  name:                         "" => "tfbug-bastion-subnet"
  network_acl:                  "" => "<computed>"
  resource_controller_url:      "" => "<computed>"
  resource_crn:                 "" => "<computed>"
  resource_group_name:          "" => "<computed>"
  resource_name:                "" => "<computed>"
  resource_status:              "" => "<computed>"
  status:                       "" => "<computed>"
  total_ipv4_address_count:     "" => "256"
  vpc:                          "" => "r006-ed0e6556-7aa8-4a56-b247-094f09d1b2c4"
  zone:                         "" => "us-south-1"
ibm_is_subnet.frontend: Still creating... (10s elapsed)
ibm_is_subnet.frontend: Creation complete after 11s (ID: 0717_d54f5a9e-548a-4661-9fa6-71dd1de75062)

Error: Error applying plan:

2 errors occurred:
	* ibm_is_subnet.backend: 1 error occurred:
	* ibm_is_subnet.backend: {"errors":[{"code":"bad_field","message":"Error with Subnet_Service CreateNetwork Request: Network creation failed. Network.networking.genesis.ibm.com \"0717_d84ab06e-e93c-48ac-98c7-1189f16ba43e\" is invalid: spec.cidr: Duplicate value: \"10.240.0.0/24\". Verify request parameters against API documentation and try again.. k8errors: Network.networking.genesis.ibm.com \"0717_d84ab06e-e93c-48ac-98c7-1189f16ba43e\" is invalid: spec.cidr: Duplicate value: \"10.240.0.0/24\""}],"trace":"a6f05098-f3c5-47df-8f77-058610787ed0"}


	* ibm_is_subnet.bastion: 1 error occurred:
	* ibm_is_subnet.bastion: {"errors":[{"code":"bad_field","message":"Error with Subnet_Service CreateNetwork Request: Network creation failed. Network.networking.genesis.ibm.com \"0717_a2fba440-547f-4260-89eb-351e2adbd134\" is invalid: spec.cidr: Duplicate value: \"10.240.0.0/24\". Verify request parameters against API documentation and try again.. k8errors: Network.networking.genesis.ibm.com \"0717_a2fba440-547f-4260-89eb-351e2adbd134\" is invalid: spec.cidr: Duplicate value: \"10.240.0.0/24\""}],"trace":"6be9d0d7-5d6c-40c4-a91e-c4167d4d75bc"}
```

**Comment by powellquiring** [comment]
Likely a bug in the service: https://jiracloud.swg.usma.ibm.com:8443/browse/RNOS-1159
Is there any way dump out some API calls that would help them debug it?

**Comment by VaishnaviGopal** [comment]
Using Parallelism while creating subnets or Using "depends_on" for the creation of each of the next subnets should solve the issue. (i.e., inside the subnet.backend resource add depends_on attribute and assign it to the subnet.frontend resource and for subnet.bastion assign the depends_on to the subnet.backend)

**Comment by hkantare** [comment]
Serialized the creation of subnets on same VPC. Fix available in master branch

**Comment by VaishnaviGopal** [comment]
The issue is fixed and the updates are available in IBM Cloud Provider V0.17.6. Link to download this plugin: https://github.com/IBM-Cloud/terraform-provider-ibm/releases/tag/v0.17.6

**Comment by hkantare** [comment]
Closing the issue
